libs/libc/string/lib_strsignal.c

```diff
@@ -22,53 +22,24 @@
  * Included Files
  ****************************************************************************/
 
+#include <errno.h>
 #include <signal.h>
+#include <stdio.h>
 #include <string.h>
 
 /****************************************************************************
- * Private Data
+ * Pre-processor Definitions
  ****************************************************************************/
 
-/* We don't know what signals names will be assigned to which signals in
- * advance and we do not want to return a volatile value.  One solution is
- * this silly array of useless names:
- */
-
-static FAR const char *g_default_sigstr[32] =
-{
-  "Signal 0",
-  "Signal 1",
-  "Signal 2",
-  "Signal 3",
-  "Signal 4",
-  "Signal 5",
-  "Signal 6",
-  "Signal 7",
-  "Signal 8",
-  "Signal 9",
-  "Signal 10",
-  "Signal 11",
-  "Signal 12",
-  "Signal 13",
-  "Signal 14",
-  "Signal 15",
-  "Signal 16",
-  "Signal 17",
-  "Signal 18",
-  "Signal 19",
-  "Signal 20",
-  "Signal 21",
-  "Signal 22",
-  "Signal 23",
-  "Signal 24",
-  "Signal 25",
-  "Signal 26",
-  "Signal 27",
-  "Signal 28",
-  "Signal 29",
-  "Signal 30",
-  "Signal 31",
-};
+#ifdef CONFIG_LIBC_STRSIGNAL_SHORT
+#  define CASE_SIG_STR(sig, msg) \
```

Comment:
Could you please expand the comment here as I'm not sure if I understand it correctly?

Reply:
like strerror:

1. LIBC_STRSIGNAL=n, return "Unknown signal"
2. LIBC_STRSIGNAL=y, CONFIG_LIBC_STRERROR_SHORT=y, return "Signal xx"
3. LIBC_STRSIGNAL=y, CONFIG_LIBC_STRERROR_SHORT=n, return "SIGWORK"...

Reply:
> like strerror:
> 
> 1. LIBC_STRSIGNAL=n, return "Unknown signal"
> 2. LIBC_STRSIGNAL=y, CONFIG_LIBC_STRERROR_SHORT=y, return "Signal xx"
> 3. LIBC_STRSIGNAL=y, CONFIG_LIBC_STRERROR_SHORT=n, return "SIGWORK"...
> 

It is very similar except:
1. LIBC_STRSIGNAL=n, return "Signal X".
2. LIBC_STRSIGNAL=y, CONFIG_LIBC_STRERROR_SHORT=y, return "SIGTRAP" (for example)
3. LIBC_STRSIGNAL=y, CONFIG_LIBC_STRERROR_SHORT=n, return "Trace/breakpoint trap".

Reply:
If so, LIBC_STRSIGNAL should be n by default like LIBC_STRERROR

Reply:
In general I agree. But current implementation behaves like "LIBC_STRSIGNAL=y, CONFIG_LIBC_STRERROR_SHORT=y". So here is what I mean by talking that defaulting `LIBC_STRSIGNAL=n` will be a "breaking change". I want to notify users that expected output will be changed with default config

Reply:
@xiaoxiang781216 do you recommend to default it to `n`? Or `default DEFAULT_SMALL` for both `LIBC_STRERROR` and `LIBC_STRSIGNAL`?

Reply:
@hartmannathan @davids5 what is your opinion on this? Should we go with the smallest possible implementation for DEFAULT_SMALL?

Reply:
I personally do not like glumping a bunch of stuff under DEFAULT_SMALL. (and yes I know you can change it) 

When we run out of code space the first thing I do is CONFIG_LIBC_STRERROR=n. So my take on this is make it as configurable and small as possible.  

I think the graduality from 0 code size increase (remove the feature) terse: "sig N" to verbose "Signal Trace/breakpoint trap"  is important but who decides what comes in and how much space it should take up in their bin, is ultimately up to the individual configuring the system and we should not force any of it on anyone.  Nothing worse then running out of code space, other then then running out of code space on a Friday at 5 pm :)

Reply:
I think that the case here is that `CONFIG_LIBC_STRERROR=n` by default, so the minimal footprint is selected by default. Here before the change the string alias like `"SIGHUP"` is returned, so if we default `CONFIG_LIBC_STRSIGNAL=n` it will affect all the users. Defaulting both `CONFIG_LIBC_STRERROR` and `CONFIG_LIBC_STRSIGNAL` to `n` in DEFAULT_SMALL case make impact a bit less as only configurations with DEFAULT_SMALL will be affected.